include/openssl/thread.h

```diff
@@ -59,6 +59,10 @@
 
 #include <sys/types.h>
 
+#if defined(__OpenBSD__)
+#   include <pthread.h>
+#endif
```

Comment:
Nit: This should be moved inside the `!defined(__GLIBC__)` block to respect `OPENSSL_THREADS`.

Reply:
Agree.

This leaks `<pthread.h>` everywhere since `thread.h` is a public header... Not sure I like that...

Reply:
Oof, I hadn't considered that. Thanks for the pointer. How does the updated diff look to you?